### PR TITLE
Operator names

### PR DESCRIPTION
**Pull request by tuschcarsten**
In an old version of the specification, operators were defined by using a special syntax with operator+, operator- etc.
In the current version of the specification, this has been changed to normal method names, using underscores to indicate that those methods are special (etc. _add_). This decision was made by Christopher, Berthold and me, due to the lack of other remaining project members who wanted to work on the specification. The good thing about this is that a special case in the AST generation can be omitted (checking whether the method has an identifier or is an operator).

### CODE REVIEW

File: src/test/resources/testPrograms/expressions/SelfDefinedOperators.output [hunk 1]
```diff
@@ -1 +1 @@
-operator_unary-operator+operator-operator*operator/operator%operator<operator>operator<=operator>=operator=operator!=operator_notoperator_xoroperator_andoperator_or
\ No newline at end of file
+operator_unary-operator+operator-operator*operator/operator%operator<operator>operator<=operator>=operator=operator!=operator_notoperator_andoperator_or
```
Comment (lummax) [inline src/test/resources/testPrograms/expressions/SelfDefinedOperators.output]: Shouldn't you change the output strings aswell? Although it's not necessary.

File: src/main/java/de/uni/bremen/monty/moco/util/OperatorMapper.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+ * to do so, delete this exception statement from your version.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this library.
+ */
+package de.uni.bremen.monty.moco.util;
+
+public class OperatorMapper {
+	public static String methodFromBinaryOperator(String operator) {
+		switch (operator) {
```
Comment (lummax) [inline src/main/java/de/uni/bremen/monty/moco/util/OperatorMapper.java]: Isn't a map way easier?

**Comment by lummax** [comment]
If you rebase against moco/master you should be able to loose the merge commit. But it's not important.

**Comment by tuschcarsten** [comment]
Tnanks for your reply. I changed the output of the test-case and removed the OperatorMapper class. Instead the ASTBuilder now contains a Map<String, String> for unary and binary operator mappings.

I will rebase this branch if the pull request is accepted (I will then merge it)  in order to not destroy the history of this pull request, but keep the history of the repository clean.

**Pull request merged**